test/integration/govuk_index/updating_popularity_data_test.rb

```diff
@@ -0,0 +1,42 @@
+require 'integration_test_helper'
+
+class GovukIndex::UpdatingPopularityDataTest < IntegrationTest
+  def test_updates_the_popularity_when_it_exists
+    insert_document('govuk_test', { link: '/test', popularity: 0.3 }, id: '/test', type: 'edition')
+    commit_index("govuk_test")
+
+    document_count = 4
+    document_rank = 2
+    insert_document("page-traffic_test", { rank_14: document_rank, path_components: ['/test'] }, id: '/test', type: "page-traffic")
+    setup_page_traffic_data(document_count: document_count)
+
+    popularity = 1.0 / ([document_rank, document_count].min + SearchConfig.instance.popularity_rank_offset)
+
+    GovukIndex::PopularityUpdater.update('govuk_test')
+
+    assert_document_is_in_rummager({ 'link' => '/test', 'popularity' => popularity }, type: 'edition', index: 'govuk_test')
+  end
+
+  def test_set_the_popularity_to_the_lowest_popularity_when_it_doesnt_exist
+    insert_document('govuk_test', { link: '/test', popularity: 0.3 }, id: '/test', type: 'edition')
+    commit_index("govuk_test")
+
+    document_count = 4
+    setup_page_traffic_data(document_count: document_count)
+
+    popularity = 1.0 / (document_count + SearchConfig.instance.popularity_rank_offset)
+
+    GovukIndex::PopularityUpdater.update('govuk_test')
+
+    assert_document_is_in_rummager({ 'link' => '/test', 'popularity' => popularity }, type: 'edition', index: 'govuk_test')
+  end
+
+  def test_correct_copies_version_information
```

Comment:
The name of this test is a bit clunky. Maybe just drop the "correct"?

Also, do you think it's worth testing the race condition where the version has been increased by a concurrent publishing update?